### PR TITLE
Intro tamper events endpoint

### PR DESCRIPTION
**Pull request by smfarrelly**

### CODE REVIEW

File: nexus_api_open_api_spec.yaml [hunk 1]
```diff
@@ -100,6 +116,95 @@ tags:
...
+                "$ref": "#/components/schemas/TamperEvents"
+        '400':
+          "$ref": "#/components/replies/400Response"
+        '401':
+          "$ref": "#/components/replies/401Response"
+        '403':
+          "$ref": "#/components/replies/403Response"
+        '404':
+          "$ref": "#/components/replies/404Response"
+    put:
```
Comment (ricehornet) [inline nexus_api_open_api_spec.yaml]: I think this `PUT` is for the `tamper_event` endpoint, not `tamper_events`.

Comment (smfarrelly) [reply nexus_api_open_api_spec.yaml]: I do not think we need separate endpoints. My preference is for a single plural endpoint.

That being said this should probably be a POST method as opposed to PUT.

Comment (ricehornet) [reply nexus_api_open_api_spec.yaml]: OK for using a single endpoint, like we do for usage data. Should the input then be an array of `TamperEvent`s?

My understanding of POST is that it's not idempotent, but PUT is. I think this is just like usage data, so should be PUT.

Comment (smfarrelly) [reply nexus_api_open_api_spec.yaml]: An array of TamperEvents makes sense to me.

My understanding matches yours, which is why I think we _should_ be using `POST` as the user will not be specifying an identifier. I thin usage_data could be updated to use `POST` as well.

Comment (chadnorvell) [reply nexus_api_open_api_spec.yaml]: I agree with `POST`ing an array. Plus, `PUT` implies you can mutate events that already exist which I don't believe we want.

**Comment by ricehornet** [comment]
OK, I have made the changes as per the comments above. Going to get this in as the rest looks OK to me.

**Pull request merged**